server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -523,6 +525,20 @@ public String getHeader(String header) {
         return headers.get(header);
     }
 
+    public void addQueryGroupHeadersTo(final ThreadContext threadContext) {
```

Comment:
`task.addQueryGroupHeaders` seems simple enough, addQueryGroupHeaders`To` seems a nit abrupt or confusing in some case that query group headers are added to ThreadContext (the function prameter)